wcivf/tests/test_urls.py

```diff
@@ -0,0 +1,51 @@
+import pytest
+from dc_utils.tests.helpers import validate_html
+from django.urls import reverse
+from people.tests.factories import PersonFactory, PersonPostFactory
+from parties.tests.factories import PartyFactory
+from elections.tests.factories import (
+    ElectionFactory,
+    PostElectionFactory,
+)
+
+
+class TestHtml:
+    @pytest.fixture
+    def urls(self):
+        return [
+            reverse("home_view"),
+            reverse("about_view"),
+            reverse("standing_as_a_candidate"),
+            reverse("opensearch"),
+            reverse("status_check_view"),
+            reverse("privacy_view"),
+            PersonPostFactory(
+                election=ElectionFactory()
+            ).person.get_absolute_url(),
+            reverse(
+                "email_person_view",
+                kwargs={"pk": PersonFactory().pk, "ignored_slug": "sadiq-khan"},
+            ),
+            reverse("parties_view"),
+            PartyFactory().get_absolute_url(),
+            reverse("elections_view"),
+            ElectionFactory().get_absolute_url(),
+            PostElectionFactory().get_absolute_url(),
+            reverse("postcode_view", kwargs={"postcode": "TE11ST"}),
+            reverse("results_list_view"),
+            reverse("api:api-root"),
+            reverse("dc_signup_form:mailing_list_signup_view"),
+            reverse("feedback_form_view"),
+            reverse("sv_voting_system_view"),
+            reverse("fptp_voting_system_view"),
+            reverse("ams_voting_system_view"),
+        ]
+
+    @pytest.mark.django_db
+    def test_html_valid(self, client, subtests, urls):
+        for url in urls:
+            with subtests.test(msg=url):
+                assert client.get(url).status_code == 200 or 301
```

Comment:
Are some of these urls still returning a 301 status code? If so, then those should be removed rather than changing the assert here. As remember the aim of this test is to run checks against the HTML  - if a response is a 301, it won't contain any HTML so we shouldn't test it. 

Unless we are getting a 301 where we expect a 200 status code. In that case we may be getting redirected because something is going wrong with the objects that we are setting up.

Reply:
`/privacy` is redirecting to the DC website. I assumed in that case a 301 is expected, but now that I look again, `/about` (which is also redirecting) isn't generating a 301.

Reply:
When I print out the content for `/privacy`, this is what I get: 
`b''`

Reply:
Oh sorry either a 301 or 302 (both redirect codes) we shouldn't test as the response won't include any HTML to validate